lib/vm/src/instance.rs

```diff
@@ -1145,7 +1145,7 @@ unsafe fn get_memory_slice<'instance>(
         let import = instance.imported_memory(init.location.memory_index);
         import.definition.as_ref().clone()
     };
-    slice::from_raw_parts_mut(memory.base, memory.current_length)
+    slice::from_raw_parts_mut(memory.base, memory.current_length.try_into().unwrap())
```

Comment:
Converting u32 to usize in a failable way is probably overkill -- I don't think Wasmer is likely to run on any platform where a u32 can't always convert into a usize.  This is consistent with some other code I've seen in the wasmer code base though so you don't have to change it, it just seems a bit odd to me

Reply:
It's what the compiler error recommended. I have a slight preference to avoid `as` casting (very convenient, but also seems like a quick way to get corner-case bugs).